media/timestamp_corrector_test.go

```diff
@@ -108,3 +117,200 @@ func TestTimestampCorrector_Burst(t *testing.T) {
 	out3 := tc.Process(ctx, ts3)
 	assert.Equal(ts3, out3, "post-flush normal call changed ts")
 }
+
+func TestTimestampCorrector_WithMatchingUA(t *testing.T) {
+	// UA is Safari on MacOS - should trigger timestamp correction
+	userAgent := "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.5 Safari/605.1.15"
+	fps := 30.0
+	ctx := context.Background()
+	assert := assert.New(t)
+
+	// simulate two packets spaced normally but with an inflated tsDelta to trigger fix
+	t0 := time.Date(2025, 5, 30, 0, 0, 0, 0, time.UTC)
+	dtNs := time.Duration((1.0/fps)*1e9) * time.Nanosecond
+	t1 := t0.Add(dtNs)
+
+	tc := NewTimestampCorrector(TimestampCorrectorConfig{
+		FPS:       fps,
+		Clock:     makeClock([]time.Time{t0, t1}),
+		UserAgent: userAgent,
+	})
+
+	ts1 := int64(1000)
+	out1 := tc.Process(ctx, ts1)
+	assert.Equal(ts1, out1, "first call changed ts")
+
+	// choose a delta large enough: > thresholdFreq*dt
+	dtSec := dtNs.Seconds()
+	largeDelta := int64((tc.thresholdFreq * dtSec) + 2)
+	ts2 := ts1 + largeDelta
+	out2 := tc.Process(ctx, ts2)
+	expected2 := multiplyAndDivide(ts2, 90_000, 1_000_000)
+	assert.Equal(expected2, out2, "bug-clock second call")
+
+	// subsequent calls also fixed
+	ts3 := ts2 + 20000
+	out3 := tc.Process(ctx, ts3)
+	expected3 := multiplyAndDivide(ts3, 90_000, 1_000_000)
+	assert.Equal(expected3, out3, "subsequent bug-clock call")
+
+}
+
+func TestTimestampCorrector_WithNonMatchingUA(t *testing.T) {
+	// UA is chrome on MacOS
+	// Should *not* trigger timestamp correction even if it's detected as being broken
+	userAgent := "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/137.0.0.0 Safari/537.36"
+	fps := 30.0
+	ctx := context.Background()
+	assert := assert.New(t)
+
+	// simulate two packets spaced normally but with an inflated tsDelta to trigger fix
+	t0 := time.Date(2025, 5, 30, 0, 0, 0, 0, time.UTC)
+	dtNs := time.Duration((1.0/fps)*1e9) * time.Nanosecond
+	t1 := t0.Add(dtNs)
+
+	tc := NewTimestampCorrector(TimestampCorrectorConfig{
+		FPS:       fps,
+		Clock:     makeClock([]time.Time{t0, t1}),
+		UserAgent: userAgent,
+	})
+
+	ts1 := int64(1000)
+	out1 := tc.Process(ctx, ts1)
+	assert.Equal(ts1, out1, "first call changed ts")
+
+	// choose a delta large enough: > thresholdFreq*dt
+	dtSec := dtNs.Seconds()
+	largeDelta := int64((tc.thresholdFreq * dtSec) + 2)
+	ts2 := ts1 + largeDelta
+	out2 := tc.Process(ctx, ts2)
+	expected2 := ts2
+	assert.Equal(expected2, out2, "bug-clock second call")
+
+	// subsequent calls also fixed
```

Comment:
nit: maybe update this comment as the correction is disabled for this test case

Reply:
Good catch, bad copy pasta ... I went ahead and updated those tests to be tabular in https://github.com/livepeer/go-livepeer/pull/3620/commits/7382797ca4bddbec43fbfe98b8c0c077d6793c2d so there's less repetition